

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Getting started &mdash; scGNN 0.1 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Data Processing" href="Data%20Processing.html" />
    <link rel="prev" title="Contact" href="Contact.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home" alt="Documentation Home"> scGNN
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Main</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="About.html">About</a></li>
<li class="toctree-l1"><a class="reference internal" href="Installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="Quickstart.html">Quickstart</a></li>
<li class="toctree-l1"><a class="reference internal" href="Release.html">Release</a></li>
<li class="toctree-l1"><a class="reference internal" href="References.html">References</a></li>
<li class="toctree-l1"><a class="reference internal" href="Contact.html">Contact</a></li>
</ul>
<p class="caption"><span class="caption-text">Tutorials</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="Data%20Processing.html">Data Processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="Downstream%20Analysis.html">Downstream Analysis</a></li>
</ul>
<p class="caption"><span class="caption-text">Example Datasets</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Benchmark%20data.html">Benchmark data</a></li>
<li class="toctree-l1"><a class="reference internal" href="Casestudy.html">Casestudy</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">scGNN</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>Getting started</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/Getting started.md.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="getting-started">
<h1>Getting started<a class="headerlink" href="#getting-started" title="Permalink to this headline">¶</a></h1>
<div class="section" id="preprocess">
<h2>Preprocess<a class="headerlink" href="#preprocess" title="Permalink to this headline">¶</a></h2>
<p>Datasets should be preprocessed to proceed. scGNN accepts scRNA-seq data format: CSV and 10X</p>
<div class="section" id="prepare-datasets">
<h3>1. Prepare datasets<a class="headerlink" href="#prepare-datasets" title="Permalink to this headline">¶</a></h3>
<p>CSV format explicitly shows data in plain text. We take an example of Alzheimer’s disease datasets (<a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE138852">GSE138852</a>) analyzed in the manuscript.</p>
<p>10X format stores scRNA-seq expression sparsely, so is commonly used for huge datasets. Take an example of <a class="reference external" href="https://data.humancellatlas.org/explore/projects/4d6f6c96-2a83-43d8-8fe1-0f53bffd4674">liver cellular landscape study</a> from human cell atlas(<a class="reference external" href="https://data.humancellatlas.org/">https://data.humancellatlas.org/</a>)</p>
<div class="section" id="example">
<h4>Example:<a class="headerlink" href="#example" title="Permalink to this headline">¶</a></h4>
<div class="section" id="csv-format">
<h5>CSV format<a class="headerlink" href="#csv-format" title="Permalink to this headline">¶</a></h5>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>mkdir GSE138852
wget -P GSE138852/ https://ftp.ncbi.nlm.nih.gov/geo/series/GSE138nnn/GSE138852/suppl/GSE138852_counts.csv.gz
</pre></div>
</div>
</div>
<div class="section" id="x-format">
<h5>10X format<a class="headerlink" href="#x-format" title="Permalink to this headline">¶</a></h5>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>mkdir liver
wget -P liver https://data.humancellatlas.org/project-assets/project-matrices/4d6f6c96-2a83-43d8-8fe1-0f53bffd4674.homo_sapiens.mtx.zip
<span class="nb">cd</span> liver
unzip 4d6f6c96-2a83-43d8-8fe1-0f53bffd4674.homo_sapiens.mtx.zip
<span class="nb">cd</span> ..
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="preprocess-input-files-for-scgnn">
<h3>2. Preprocess input files for scGNN<a class="headerlink" href="#preprocess-input-files-for-scgnn" title="Permalink to this headline">¶</a></h3>
<p>Program <em><strong>PreprocessingscGNN.py</strong></em> is used to generate input files for scGNN. This step generates Use_expression.csv (preprocessed file) and gets discretized regulatory signals as ltmg.csv from Left-Trunctruncated-Mixed-Gaussian(LTMG) model (Optional but recommended).</p>
<p>In <em><strong>PreprocessingscGNN.py</strong></em>, usually these parameters are used:</p>
<div class="section" id="required">
<h4>Required<a class="headerlink" href="#required" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li><p><strong>datasetName</strong> defines the raw file name of scRNA-Seq</p></li>
<li><p><strong>datasetDir</strong> folder of the raw file</p></li>
<li><p><strong>LTMGDir</strong> Ouput folder of the LTMG output</p></li>
<li><p><strong>filetype</strong> defines file type (CSV or 10X(default))</p></li>
</ul>
</div>
<div class="section" id="optional">
<h4>Optional<a class="headerlink" href="#optional" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li><p><strong>transform</strong> defines the type of transformation, default is logarithm transformation. In the implementation, we use log(x+1), x is the original expression level.</p></li>
<li><p><strong>cellRatio</strong> defines the maximum ratio of zeros in cells. Default is 0.99. This parameter filters out cells with more than 99% genes that are zeros.</p></li>
<li><p><strong>geneRatio</strong> defines the maximum ratio of zeros in genes. Default is 0.99. This parameter filters out genes with more than 99% genes that are zeros.</p></li>
<li><p><strong>geneCriteria</strong> defines which criteria to select most variant genes, default is variance.</p></li>
<li><p><strong>geneSelectnum</strong> selects a number of most variant genes. The default gene number is 2000.</p></li>
</ul>
<p>The running time of <em><strong>PreprocessingscGNN.py</strong></em> is dependent on the cell number and gene numbers selected. It takes ~20 minutes (GSE138852) and ~28 minutes (liver) to generate the files needed.</p>
</div>
<div class="section" id="id1">
<h4>Example:<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h4>
<div class="section" id="id2">
<h5>CSV format<a class="headerlink" href="#id2" title="Permalink to this headline">¶</a></h5>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>python -W ignore PreprocessingscGNN.py --datasetName GSE138852_counts.csv.gz --datasetDir GSE138852/ --LTMGDir GSE138852/ --filetype CSV --geneSelectnum <span class="m">2000</span>
</pre></div>
</div>
</div>
<div class="section" id="id3">
<h5>10X format<a class="headerlink" href="#id3" title="Permalink to this headline">¶</a></h5>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>python -W ignore PreprocessingscGNN.py --datasetName 481193cb-c021-4e04-b477-0b7cfef4614b.mtx --datasetDir liver/ --LTMGDir liver/ --geneSelectnum <span class="m">2000</span>
</pre></div>
</div>
</div>
</div>
</div>
</div>
<div class="section" id="check-results">
<h2>Check Results<a class="headerlink" href="#check-results" title="Permalink to this headline">¶</a></h2>
<p>In outputdir now, we have four output files.</p>
<ul class="simple">
<li><p>*<strong>_recon.csv</strong>:        Imputed gene expression matrix. Row as gene, col as cell. First row as gene name, First col as the cell name.</p></li>
<li><p>*<strong>_embedding.csv</strong>:    Learned embedding (features) for clustering. Row as cell, col as embeddings. First row as the embedding names (no means). First col as the cell name.</p></li>
<li><p>*<strong>_graph.csv</strong>:        Learned graph edges of the cell graph in tuples: nodeA,nodeB,weights. First row as the name.</p></li>
<li><p>*<strong>_results.txt</strong>:      Identified cell types. First row as the name.</p></li>
</ul>
<p>For a complete list of options provided by scGNN:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>python scGNN.py --help
</pre></div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="Data%20Processing.html" class="btn btn-neutral float-right" title="Data Processing" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Contact.html" class="btn btn-neutral float-left" title="Contact" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, Juexin Wang, Jianting Gong

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>